pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/column/PhysicalColumnIndexContainer.java

```diff
@@ -115,6 +96,8 @@ public PhysicalColumnIndexContainer(SegmentDirectory.Reader segmentReader, Colum
     if (loadTextIndex) {
       Preconditions.checkState(segmentReader.hasIndexFor(columnName, ColumnIndexType.TEXT_INDEX));
       Map<String, Map<String, String>> columnProperties = indexLoadingConfig.getColumnProperties();
+      // it's awkward to model the distinction between this and the FST index, which may replace the implemention,
```

Comment:
FST and text index are used for different queries:
- FST - for LIKE and REGEXP_LIKE
- Text - for TEXT_MATCH

For now we should still provide 2 separate index readers for them. FST index won't be able to replace the text index